client/analytics/components/report-chart/index.js

```diff
@@ -180,7 +199,16 @@ ReportChart.propTypes = {
 
 export default compose(
 	withSelect( ( select, props ) => {
-		const { query, endpoint } = props;
+		const { query, endpoint, filters } = props;
+		const chartMode = props.mode || getChartMode( filters, query ) || 'time-comparison';
```

Comment:
Now that we are getting the chart mode inside `withSelect`, I think it would make sense to overwrite the `mode` prop at this step.

This way, inside the render method we don't need to compute the `chartMode` again, and we don't have different values for `mode` and `chartMode`, which might be confusing.

Would that work?

Reply:
Good catch - updated in https://github.com/woocommerce/wc-admin/pull/1391/commits/275356e15585f88dc9f21d0a676dc6e111633c0d.